runtime/domain.c

```diff
@@ -1272,6 +1269,7 @@ static void domain_terminate()
     handover_ephemerons(domain_state);
     handover_finalisers(domain_state);
 
+    CAML_EVENTLOG_TEARDOWN();
```

Comment:
In this loop, we may still take part in STW sections.

I think we should put the `CAML_EVENTLOG_TEARDOWN` after the domain is no longer part of STW sections; say after `caml_teardown_major_gc`